src/vaadin-combo-box-dropdown.html

```diff
@@ -249,9 +249,26 @@
         this._translateY = Math.round(this._translateY * _devicePixelRatio) / _devicePixelRatio;
         this.$.overlay.style.transform = `translate3d(${this._translateX}px, ${this._translateY}px, 0)`;
 
-        this.$.overlay.style.width = this.positionTarget.clientWidth + 'px';
         this.$.overlay.style.justifyContent = this.alignedAbove ? 'flex-end' : 'flex-start';
 
+        const customOverlayWidth = window.ShadyCSS ?
```

Comment:
Please consider refactoring the code in two separate methods:

```js
_getCustomWidth() {
  return window.ShadyCSS ?
    window.ShadyCSS.getComputedStyleValue(this, '--vaadin-combo-box-overlay-width') :
    getComputedStyle(this).getPropertyValue('--vaadin-combo-box-overlay-width');
}

_setOverlayWidth() {
  const inputWidth = this.positionTarget.clientWidth + 'px';
  const customWidth = this._getCustomWidth();

  if (window.ShadyCSS) {
    window.ShadyCSS.styleSubtree(this.$.overlay, {
      '--vaadin-combo-box-overlay-width': customWidth,
      '--_vaadin-combo-box-overlay-default-width': inputWidth
    });
  } else {
    this.$.overlay.style.setProperty('--_vaadin-combo-box-overlay-default-width', inputWidth);

    if (customWidth === '') {
      this.$.overlay.style.removeProperty('--vaadin-combo-box-overlay-width');
    } else {
      this.$.overlay.style.setProperty('--vaadin-combo-box-overlay-width', customWidth);
    }
  }
}
```

Reply:
Thanks for the suggestion. This looks better. Though initially I intentionally left setting the native custom props out of an else block so that the real custom props would be set on even if you have ShadyCSS enabled in Chrome or other browsers. At least when just running the demos locally with `polymer serve` even Chrome has ShadyCSS enabled by default (I guess mostly for ApplyShim) so if your code depended on the real custom properties, it would not work.

But now that I looked a bit more into this I came up with a better check.

First I though I could [feature detect CSS custom property support](https://stackoverflow.com/questions/38012009/how-to-feature-detect-for-css-custom-properties) like `window.CSS && CSS.supports('color', 'var(--primary)')` but that also won't work right since it returns true in Chrome regardless of what polyfills are loaded and then I tested that the overlay size actually breaks with some polyfill combinations with only that check. With `wc-shimcssproperties` query param Chrome also needs to use `ShadyCSS.styleSubtree`.

Then I looked at what `ShadyCSS` contains and it seems `ShadyCSS.nativeCss` is a boolean that tells whether CSS custom properties are native or shimmed so that seems like the correct check to work in all scenarios. I'll make an update using this.

Reply:
```js
if (window.ShadyCSS && !window.ShadyCSS.nativeCss) {
  // use ShadyCSS.styleSubtree
} else {
  // set native custom props
}
```
This seems to work nicely in both IE11 and Chrome (with different polyfill combinations).

On Chrome I tested with:
[demo/](http://0.0.0.0:3000/components/vaadin-combo-box/demo/#combo-box-basic-demos)
[demo/?wc-shadydom](http://0.0.0.0:3000/components/vaadin-combo-box/demo/?wc-shadydom#combo-box-basic-demos) (ShadyDOM without shimming custom props)
[demo/?wc-shadydom&wc-shimcssproperties](http://0.0.0.0:3000/components/vaadin-combo-box/demo/?wc-shadydom&wc-shimcssproperties#combo-box-basic-demos) (ShadyDOM and shimming custom props)

Note `wc-shimcssproperties` without `wc-shadydom` doesn't seem to work at all (all styles are broken) so I guess `wc-shimcssproperties` is only meant to be used with `wc-shadydom`.

I also briefly tried `wc-ce` with some of the others but that didn't seem to make a difference.

Reply:
This also makes it so that `ShadyCSS.styleSubtree` is not unnecessarily called on browsers that might have ApplyShim loaded but don't shim custom props (like Chrome by default when running this demo locally.

Reply:
Updated the code. PTAL.